processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/UnsafeSortDataRows.java

```diff
@@ -140,19 +143,25 @@ public void initialize() throws MemoryException, CarbonSortKeyAndGroupByExceptio
     semaphore = new Semaphore(parameters.getNumberOfCores());
   }
 
-  private UnsafeCarbonRowPage createUnsafeRowPage()
-      throws MemoryException, CarbonSortKeyAndGroupByException {
-    MemoryBlock baseBlock =
-        UnsafeMemoryManager.allocateMemoryWithRetry(this.taskId, inMemoryChunkSize);
-    boolean isMemoryAvailable =
-        UnsafeSortMemoryManager.INSTANCE.isMemoryAvailable(baseBlock.size());
-    if (isMemoryAvailable) {
-      UnsafeSortMemoryManager.INSTANCE.allocateDummyMemory(baseBlock.size());
-    } else {
-      // merge and spill in-memory pages to disk if memory is not enough
-      unsafeInMemoryIntermediateFileMerger.tryTriggerInmemoryMerging(true);
+  private UnsafeCarbonRowPage createUnsafeRowPage() {
+    try {
+      MemoryBlock baseBlock =
+          UnsafeMemoryManager.allocateMemoryWithRetry(this.taskId, inMemoryChunkSize);
+      boolean isMemoryAvailable =
+          UnsafeSortMemoryManager.INSTANCE.isMemoryAvailable(baseBlock.size());
+      if (isMemoryAvailable) {
+        UnsafeSortMemoryManager.INSTANCE.allocateDummyMemory(baseBlock.size());
+      } else {
+        // merge and spill in-memory pages to disk if memory is not enough
+        unsafeInMemoryIntermediateFileMerger.tryTriggerInmemoryMerging(true);
+      }
+      return new UnsafeCarbonRowPage(tableFieldStat, baseBlock, !isMemoryAvailable, taskId);
+    } catch (MemoryException | CarbonSortKeyAndGroupByException e) {
```

Comment:
Here we are eating up what type of exception was thrown

Reply:
But caller always throw CarbonSortKeyAndGroupByException.

but still modified code as per suggestion.